### PR TITLE
Workflow has been renamed to SwiftCurrent!  All the same greatness, but more discoverable and less confusing.

### PR DESCRIPTION
**Pull request by Richard-Gist**
## Checklist:
- [x] Is the linter reporting 0 errors?
- [x] Did you comply with our [styleguide](https://github.com/wwt/Workflow/blob/main/STYLEGUIDE.md)?
- [x] Is there [adequate test coverage](https://github.com/wwt/Workflow/blob/main/CONTRIBUTING.md#test-etiquette) for your new code?
- [x] Does the CI pipeline pass?
- [x] Is the [static analysis tool](https://sonarcloud.io/dashboard?id=wwt_Workflow) reporting 0 issues?
- [x] Did you [update the documentation](https://github.com/wwt/Workflow/blob/main/CONTRIBUTING.md#documentation)?
- [x] Did you [update the sample app](https://github.com/wwt/Workflow/blob/main/CONTRIBUTING.md#sample-app)?
- [x] Do we need to [increment the minor/major version](https://github.com/wwt/Workflow/blob/main/CONTRIBUTING.md#versioning)?
- [x] Did you [change the public API](https://github.com/wwt/Workflow/blob/main/CONTRIBUTING.md#public-api)?
- [x] Have you done everything you can to make sure that errors that can occur are compile-time errors, and if they have to be runtime do you have adequate tests and documentation around those runtime errors? [For more details](https://github.com/wwt/Workflow/blob/main/CONTRIBUTING.md#errors).

----

### If Applicable:
- [ ] Did you test when the first item is skipped?
- [ ] Did you test when the last item is skipped?
- [ ] Did you test when middle items are skipped?
- [ ] Did you test when incorrect data is passed forward?
- [ ] Did you test proceeding backwards?

----

### If Public API Has Changed:
- [ ] Did you deprecate (rather than remove) any old methods/variables/etc? [Our philosophy for deprecation](https://github.com/wwt/Workflow/blob/main/CONTRIBUTING.md#deprecation).
- [ ] Have you done the best that you can to make sure that the compiler guides people to changing to the new API? (Example: the renamed attribute)
- [x] If necessary, have you tested the upgrade path for at least N-1 versions? For example, if data persists between v1 and v2 then that upgrade should be tested and as easy as we can make it.

### CODE REVIEW

**Comment by Richard-Gist** [comment]
EOD: @Tyler-Keith-Thompson got me to start renaming the app because "of course we'll be able to do it quickly".  Either way we should move quickly to publish the name change.

We pondered between `SwiftCurrent_UIKit` and `SwiftCurrentUIKit` and we ultimately decided `SwiftCurrent_UIKit` for readability.  It becomes more apparent when you think of the next module: `SwiftCurrent_SwiftUI`.

**Comment by Tyler-Keith-Thompson** [comment]
CI/CD is passing again and everything code-wise has been renamed. SonarCloud is reporting successfully, but it's unhappy about all the changes and has regenerated all the issues we already marked as resolved.

File: UPGRADE_PATH.md [hunk 1]
```diff
@@ -1,6 +1,12 @@
 ### Upgrade Paths:
 Use this document to help you understand how to update between major versions of the library. We test upgrade paths when we ramp major versions and do our best to have the compiler help make a useful upgrade experience.
 
+<details>
+  <summary><b>V3 -> V3</b></summary>
+  update imports and pod or SPM
```
Comment (Richard-Gist) [inline UPGRADE_PATH.md:6]: We can't forget to update this.

File: UPGRADE_PATH.md [hunk 2]
```diff
@@ -1,6 +1,28 @@
...
 Use this document to help you understand how to update between major versions of the library. We test upgrade paths when we ramp major versions and do our best to have the compiler help make a useful upgrade experience.
 
+<details>
+  <summary><b>V3 -> V4</b></summary>
+
+  ### Name Change
+  The library changed its name from "Workflow" to "SwiftCurrent". This change was an important step for us, because it helps with SEO and gives people a sense that the library isn't just some generic thing slapped together. 
+
+  **If you're using CocoaPods**
+  Update your `podfile` to include `SwiftCurrent`
```
Comment (Richard-Gist) [inline UPGRADE_PATH.md:11]:
```suggestion
  Update your `Podfile` to include `SwiftCurrent`
```

File: UPGRADE_PATH.md [hunk 3]
```diff
@@ -1,6 +1,28 @@
...
+  ```ruby
+  pod 'SwiftCurrent'
```
Comment (Richard-Gist) [inline UPGRADE_PATH.md:13]: This doesn't align with our installation instructions.  #36 has some renames in the installation instructions... I'm not sure if this should be added to the installation instructions or if we just call out that you should be changing instances of `Workflow` to `SwiftCurrent`?

@morganzellers thoughts?

File: UPGRADE_PATH.md [hunk 4]
```diff
@@ -1,6 +1,28 @@
...
+  ```
+
+  **If you're using Swift Package Manager**
+  Update your URL in Xcode or your `Package.swift` file to the new one, for example:
+  ```swift
+  .package(url: "https://github.com/wwt/SwiftCurrent.git", .upToNextMajor(from: "4.0.0")),
+  ```
+
```
Comment (Richard-Gist) [inline UPGRADE_PATH.md:21]: This hit most of the upgrade but we still need to call out the import changes.

**Review by Richard-Gist — COMMENTED** [review]
just marking that I viewed this again.

File: UPGRADE_PATH.md (hunk 3 above)
Comment (morganzellers) [reply UPGRADE_PATH.md:13]: I think calling out to change instances of `Workflow` in the Podfile to `SwiftCurrent` is good, that way it keeps us out of the specification of the `UIKit` vs `Core` usage

**Review by wiemerm — APPROVED** [review]

**Review by morganzellers — APPROVED** [review]

**Pull request merged**